### PR TITLE
wait-online: support alternative interface names

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
cc @mrc0mmand

**Comment by yuwata** [comment]
Yay, all tests are passed in CentOS(arch).

File: test/test-network/conf/25-default.link [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: MIT-0
```
Comment (mrc0mmand) [inline test/test-network/conf/25-default.link]: I wonder - is this supposed to be the exact copy of the `99-default.link` file? I noticed there's an extra `mac` in the `AlternativeNamesPolicy=` value, but if it's supposed to be the copy of `99-default.link` wouldn't it be better to create a relative symlink to the original file (`ln -srv network/99-default.link test/test-network/conf/25-default.link`) so it's always up to date?

Comment (yuwata) [reply test/test-network/conf/25-default.link]: I intentionally added `mac`. To detect issues like #25181. But, I forgot to add a testcase. I will add it.

Comment (yuwata) [reply test/test-network/conf/25-default.link]: I noticed that the MAC address based naming is not allowed for virtual interfaces.
Now, the file is a symlink to `network/99-default.link`.

File: src/network/wait-online/link.c [hunk 2]
```diff
@@ -81,6 +82,18 @@ static int link_update_name(Link *l, sd_netlink_message *m) {
         if (streq(ifname, l->ifname))
                 return 0;
 
+        /* kernel sometimes send wrong ifname change. Confirm the sent name. */
```
Comment (bluca) [inline src/network/wait-online/link.c]: "The kernel sometimes sends the wrong ifname change. Confirm the received name."

File: src/network/wait-online/link.c [hunk 3]
```diff
@@ -107,6 +112,39 @@ static int link_update_name(Link *l, sd_netlink_message *m) {
...
+        _cleanup_strv_free_ char **altnames = NULL;
+        int r;
+
+        assert(l);
+        assert(l->manager);
+        assert(m);
+
+        r = sd_netlink_message_read_strv(m, IFLA_PROP_LIST, IFLA_ALT_IFNAME, &altnames);
+        if (r == -ENODATA)
+                /* The message does not have IFLA_PROP_LIST container attribute. It does not means the
```
Comment (bluca) [inline src/network/wait-online/link.c]: "It does not mean"

**Review by bluca — APPROVED** [review]

**Review by mrc0mmand — APPROVED** [review]
Also LGTM, thanks!

**Pull request merged**